source/geometry_model/chunk.cc

```diff
@@ -408,9 +660,21 @@ namespace aspect
     double
     Chunk<dim>::depth(const Point<dim> &position) const
     {
+      // depth is defined wrt the reference surface point2[0]
+      // positive topography therefore has a negative depth and vv.
       return std::min (std::max (point2[0]-position.norm(), 0.), maximal_depth());
     }
 
+    template <int dim>
+    double
+    Chunk<dim>::depth_wrt_topo(const Point<dim> &position) const
+    {
+      // depth is defined wrt the reference surface point2[0] + the topography
+      // depth is therefore always positive
+      const double outer_radius = manifold.get_radius(position);
+      const Point<dim> rtopo_phi_theta = manifold.pull_back_sphere(position);
+      return std::max(0.0, outer_radius - rtopo_phi_theta[0]);
```

Comment:
Depth here is computed as the difference between the radius of modified surface (given by get_radius) and the modified point (given by pull_back_sphere).